### DISCUSSION TITLE
How can I persist the sortBy state of a Table between mounts?

### DISCUSSION THREAD

**Discussion by mperrando** [buildo/bento-design-system#826] [Q&A] [answered]
I tried to put hook on the `onSort` where I receive the current sort configuration  (`Array<SortingRule<C>>`) to save it, so that, when component will be mounted again, I can pass it to the `initialSorting` property.

But, doing so, I lose the feature of the table to sort the rows by itself, and I don't want to lose it!

I cannot either find a `ref` of the `Table` to invoke a `getCurrentSort`, or similar, to save the sort configuration when component is unmounted.

What am I missing?

Thank you!

**Comment by veej** [comment]
Passing the `onSort` function makes the sorting "controlled", meaning that you're in charge of sorting the data before passing them to the `Table`.
This behavior may be improved to address your use case but, currently, you should consider storing the sorting rules in a local state and utilize a library such as `lodash.orderby` to sort the data manually. An example can be seen [here](https://github.com/buildo/bento-design-system/blob/main/packages/bento-design-system/stories/Components/Table.stories.tsx#L536)

**Reply by mperrando** [reply]
Thank you @veej .

I think I did not myself clear enough.

I do not need to do a custom sort, I am very happy with the sort feature of the table.

I need to restore the "sort state" of the table (i.e. the little arrows in the headers the columns) between mounts or between application runs.

If I use a custom sort, the arrows in the headers reflect the sort I am applying.

**Reply by veej** [reply, marked as answer]
I understood your use case.
My solution at the moment is:
1) use `onSort` to store the current sorting, e.g. on the parent component (since you want to persist it during unmounts / remounts)
2) pass the initialSorting for when the Table is mounted again
3) use `lodash.orderby` to sort the `data` based on the current sorting stored on point 1

I am missing something? Isn't this solution working as you expect?

**Reply by mperrando** [reply]
Thank you. I let you know.
